lighthouse/multi-cluster-service-selection.md

```diff
@@ -0,0 +1,71 @@
+# Multi-Cluster Service Selection Optimization - EP
+
+## Motivation
+
+While the world is moving towards micro-services application based architecture, more organizations combine multi-cluster and multi-cloud providers for redundancy, performance, management, and other reasons.
+We would like to optimize the total cost of a Kubernetes deployment in means of communication between multi-clusters, given that we could decide between different approaches. The three approaches possible, currently, are optimizing the full service chain of a single app/deployment, multiple apps/deployments, or a smart local optimization per cluster point of view, without building the full service chain.
+In this Enhancement Proposal, we shall introduce questions, ideas, and requirements to optimize the service selection mechanism of the Lighthouse Plugin.
+
+
+### Requirements
+1. Topology:
+    1. Selection Decision should be made locally, i.e per cluster - decentralized solution.
+    2. The solution must be adaptive for topology changes.
+1. Cost Parameters/Metrics:
+    1. Latency
+    2. RTT
+    3. Networking cost
+    4. Network congestion

```

Comment:
How would you measure this?

Reply:
For congestion we could use available bandwidth vs used bandwidth in % or something like that. But really hard to measure over clusters. I'd leave that as a suggestion for future exploration, I agree with mike it's probably not doable, or not easy (unless you have full control of all the intermediate network infrastructure).